File: airbyte-persistence/job-persistence/src/main/java/io/airbyte/persistence/job/errorreporter/LoggingJobErrorReportingClient.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.persistence.job.errorreporter;

import edu.umd.cs.findbugs.annotations.Nullable;
import io.airbyte.config.FailureReason;
import io.airbyte.config.StandardWorkspace;
import java.util.Map;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Log job error reports.
 */
@SuppressWarnings("PMD.AvoidDuplicateLiterals")
public class LoggingJobErrorReportingClient implements JobErrorReportingClient {

  private static final Logger LOGGER = LoggerFactory.getLogger(LoggingJobErrorReportingClient.class);

  @Override
  public void reportJobFailureReason(@Nullable final StandardWorkspace workspace,
                                     final FailureReason reason,
                                     final String dockerImage,
                                     final Map<String, String> metadata,
                                     @Nullable final AttemptConfigReportingContext attemptConfig) {
    LOGGER.info(
        "Report Job Error -> workspaceId: {}, dockerImage: {}, failureReason: {}, metadata: {}, state: {}, sourceConfig: {}, destinationConfig: {}",
        workspace != null ? workspace.getWorkspaceId() : "null",
        dockerImage,
        reason,
        metadata,
        attemptConfig != null ? attemptConfig.state() : "null",
        attemptConfig != null ? attemptConfig.sourceConfig() : "null",
        attemptConfig != null ? attemptConfig.destinationConfig() : "null");
  }

}

File: airbyte-persistence/job-persistence/src/main/kotlin/io/airbyte/persistence/job/errorreporter/LoggingJobErrorReportingClient.kt
/*
 * Copyright (c) 2020-2026 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.persistence.job.errorreporter

import io.airbyte.config.FailureReason
import io.airbyte.config.StandardWorkspace
import io.github.oshai.kotlinlogging.KotlinLogging

/**
 * Log job error reports.
 */
class LoggingJobErrorReportingClient : JobErrorReportingClient {
  override fun reportJobFailureReason(
    workspace: StandardWorkspace?,
    reason: FailureReason,
    dockerImage: String?,
    metadata: Map<String?, String?>?,
    attemptConfig: AttemptConfigReportingContext?,
  ) {
    var err =
      mapOf(
        "workspaceId" to workspace?.workspaceId,
        "dockerImage" to dockerImage,
        "metadata" to metadata,
        "state" to attemptConfig?.state,
        "sourceConfig" to attemptConfig?.sourceConfig,
        "destinationConfig" to attemptConfig?.destinationConfig,
      )
    log.info { "Report Job Error -> $err" }
  }

  companion object {
    private val log = KotlinLogging.logger {}
  }
}
